Frends.AzureBlobStorage.ListBlobsInContainer/Frends.AzureBlobStorage.ListBlobsInContainer/ListBlobsInContainer.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.ComponentModel;
+using Azure.Storage.Blobs;
+using Azure;
+using Frends.AzureBlobStorage.ListBlobsInContainer.Definitions;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.Models;
+using System.Collections.Generic;
+#pragma warning disable CS1573
+
+namespace Frends.AzureBlobStorage.ListBlobsInContainer
+{
+    public class AzureBlobStorage
+    {
+        /// <summary>
+        ///     List of blobs in container.
+        ///     [Documentation](https://tasks.frends.com/tasks#frends-tasks/Frends.AzureBlobStorage.ListBlobsInContainer)
+        /// </summary>
+        /// <param name="source">Source values.</param>
+        /// <returns>List of blobs in container with chosen listing structure.</returns>
+        /// <exception cref="Exception">Authentication exception.</exception>
+
+        public static async Task<Result> ListBlobsInContainer([PropertyTab] Source source)
+        {
+            BlobContainerClient blobContainerClient = null;
+            var blobsList = new List<Result>();
+            var authSas = false;
+
+            switch (source.AuthenticationMethod)
+            {
+                case AuthenticationMethod.Connectionstring:
+                    if (string.IsNullOrEmpty(source.ConnectionString))
+                        throw new Exception("Connection string required.");
+                    blobContainerClient = new BlobContainerClient(source.ConnectionString, source.ContainerName);
+                    break;
+
+                case AuthenticationMethod.Sastoken:
+                    if (string.IsNullOrEmpty(source.SasToken) || string.IsNullOrEmpty(source.Uri))
+                        throw new Exception("SAS Token and URI required.");
+                    authSas = true;
+                    blobContainerClient = new BlobContainerClient(new Uri($"{source.Uri}/{source.ContainerName}?"), new AzureSasCredential(source.SasToken));
+                    break;
+            }
+
+            if (source.FlatBlobListing)
```

Comment:
Now that I am looking at this - I am not at all convinced that this is a good way doing those two ways of presenting data. I even think that maybe we should simplify this as much as we can and have two tasks instead + one that is returning a flat structure and the other that would return a hierarchy. Opinions?

Reply:
My opinion: What comes to data, there are just minor differences between these 2 listing structures and code isn't complex enough to be separated into their own tasks. Problem here is/was the UX so I made some modifications:

- Listing structure is now Selector instead of Boolean, name is changed from "FlatBlobListing" to "ListingStructure" and tooltip is more (end)user-friendly.
- Listing structure type added into BlobList (Task return).